assets/app/scripts/controllers/edit/buildConfig.js

```diff
@@ -0,0 +1,561 @@
+    // selectboxes should be select the first option.
+    $scope.updateBuilderImageStreams = function(projectName, selectFirstOption) {
+      if ($scope.options.pickedBuildFromType === 'None') {
+        $scope.pickedTriggers['imageChange'] = false;
+      } else {
+        DataService.list("imagestreams", {namespace: projectName}, function(imageStreams) {
+          $scope.buildFrom.imageStreams = [];
+          $scope.buildFrom.tags = {};
+          var projectImageStreams = imageStreams.by("metadata.name");
+          if (!_.isEmpty(projectImageStreams)) {
+            angular.forEach(projectImageStreams, function(imageStream, name) {
+              var tagList = [];
+              if (imageStream.status.tags) {
+                imageStream.status.tags.forEach(function(item){
+                  tagList.push(item["tag"]);
+                });
+              }
+              $scope.buildFrom.imageStreams.push(name);
+              $scope.buildFrom.tags[name] = tagList;
+              // If ImageStream doesn't have any tags, set tag to empty string, so the user has to pick from a existing Tags.
+              if (name === $scope.options.pickedBuildFromImageStream &&_.isEmpty(tagList)) {
+                $scope.options.pickedBuildFromImageStreamTag = "";
+              }
+            });
+            // If defined Image Stream is not present in the Image Streams of picked Namespace set tag to empty string,
+            // so the user has to pick from a existing Image Streams.
+            if (!$scope.buildFrom.imageStreams.contains($scope.options.pickedBuildFromImageStream)) {
+              $scope.options.pickedBuildFromImageStreamTag = "";
+            }
+            if (selectFirstOption) {
+              $scope.options.pickedBuildFromImageStream = $scope.buildFrom.imageStreams[0];
+              $scope.clearSelectedBuilderTag();
+            }
+          } else {
+            $scope.options.pickedBuildFromImageStream = "";
+            $scope.options.pickedBuildFromImageStreamTag = "";
+          }
+        });
+      }
+    }
+
+    $scope.clearSelectedBuilderTag = function() {
+      var tags = $scope.buildFrom.tags[$scope.options.pickedBuildFromImageStream];
+      $scope.options.pickedBuildFromImageStreamTag = _.find(tags, function(tag) { return tag == "latest" }) || tags[0] || "";
+    }
+
+    // updateOutputImageStreams creates/updates the list of imageStreams and imageStreamTags for the output image from picked namespace.
+    // As parameter takes the picked namespace, selectFirstOption as a boolean that indicates whether the imageStreams and imageStreamTags
+    // selectboxes should be select the first option.
+    $scope.updateOutputImageStreams = function(projectName, selectFirstOption) {
+      DataService.list("imagestreams", {namespace: projectName}, function(imageStreams) {
+        $scope.pushTo.imageStreams = [];
+        $scope.pushTo.tags = {};
+        var projectImageStreams = imageStreams.by("metadata.name");
```

Comment:
This chunk of code is duplicated in `$scope.updateImageSourceImageStreams()`, can it be extracted into a separate function? I'm ok with this function calling a few smaller functions.